### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "polymer-upgrade",
+  "version": "1.0.0",
```
Comment (justinfagnani) [inline package.json]: let's start with a lower, or at least pre-release version

File: package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "test:watch": "watchy -w . -i '/\\.|/node_modules|/lib/|.*\\.js$' -- npm test"
+  },
+  "keywords": [
+    "polymer",
+    "web",
+    "components"
+  ],
+  "author": "The Polymer Project Authors",
+  "license": "BSD-3-Clause",
+  "devDependencies": {
```
Comment (justinfagnani) [inline package.json]: mocha?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "require-self": "^0.1.0",
+    "typescript": "^2.0.0",
+    "watchy": "^0.6.6"
+  },
+  "dependencies": {
+    "@types/mocha": "^2.2.28"
```
Comment (justinfagnani) [inline package.json]: devDependency?

File: test/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+    "compilerOptions": {
+        "target": "es6",
+        "module": "commonjs",
+        "moduleResolution": "node",
+        "isolatedModules": false,
+        "noImplicitAny": true,
+        "removeComments": false,
+        "preserveConstEnums": true,
+        "suppressImplicitAnyIndexErrors": true
```
Comment (justinfagnani) [inline test/tsconfig.json]: noLib

File: test/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    },
+    "include": [
+        "*.ts"
```
Comment (justinfagnani) [inline test/tsconfig.json]: `src/` instead of `*.ts`? and lib.es7

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        "noImplicitAny": true,
+        "noImplicitReturns": true,
+        "noImplicitThis": true,
+        "noUnusedLocals": true,
+        "noUnusedParameters": true,
+        "strictNullChecks": true,
+        "removeComments": false,
+        "preserveConstEnums": true,
+        "suppressImplicitAnyIndexErrors": true,
+        "outDir": "./lib",
```
Comment (justinfagnani) [inline tsconfig.json]: noLib

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        "declaration": true
+    },
+    "include": [
+        "custom_typings/main.d.ts",
```
Comment (justinfagnani) [inline tsconfig.json]: add lib..es7

File: package.json (hunk 1 above)
Comment (rictic) [inline package.json]: `0.1.0` now

File: package.json (hunk 2 above)
Comment (rictic) [inline package.json]: Good catch, done

File: package.json (hunk 3 above)
Comment (rictic) [inline package.json]: Done

File: tsconfig.json (hunk 6 above)
Comment (rictic) [inline tsconfig.json]: An explicit `"lib": [ ... ]` is the ≥2.0 way of doing this. So I specified ES2016.

File: tsconfig.json (hunk 7 above)
Comment (rictic) [inline tsconfig.json]: Handled by `lib` compiler option

**Comment by rictic** [comment]
PTAL

**Comment by rictic** [comment]
Moved tests into `src/tests`, this reduces hackery by a ton. PTAL

File: tsconfig.json [hunk 8]
```diff
@@ -21,9 +21,7 @@
     },
     "include": [
         "custom_typings/main.d.ts",
-        "src/**/*.ts",
-        "src/*.ts",
-        "test/*.ts",
-        "test/**.ts"
+        "src/.ts",
```
Comment (justinfagnani) [inline tsconfig.json]: `*.ts`?

Comment (rictic) [inline tsconfig.json]: Done. Just `src/**/*.ts`

File: src/test/index.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+/// <reference path="../../node_modules/@types/mocha/index.d.ts" />
+
+import {upgrade} from '../index';
+
+describe('the upgrade function', function() {
```
Comment (justinfagnani) [inline src/test/index.ts]: let's use the TDD API

Comment (justinfagnani) [inline src/test/index.ts]:
```
--ui tdd
test/*_test.js test/**/*_test.js
```

**Comment by justinfagnani** [comment]
LGTM

please squash for the initial commit :)

File: src/test/index.ts (hunk 9 above)
Comment (rictic) [inline src/test/index.ts]: Done.

**Comment by rictic** [comment]
I'll squash, but I hope you realize the loss to future generations of precious metadata about how many times I fucked up the structure of a very basic project.

**Pull request merged**